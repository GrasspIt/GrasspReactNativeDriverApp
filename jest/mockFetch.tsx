// // add this to your setupFilesAfterEnv config in jest so it's imported for every test file
// import * as actions from './actions/oauthActions';
// async function mockFetch(url, config) {
//   switch (url) {
//     case '/oauth/token': {
//       const user = await actions.login(JSON.parse(config.body))
//       return {
//         ok: true,
//         status: 200,
//         json: async () => ({access_token}),
//       }
//     }
//     case '/user/access_token=...': {
//       const user = await actions.login(JSON.parse(config.body))
//       return {
//         ok: true,
//         status: 200,
//           json: async () => ({user}),
//         })
//       }
//     case '/user/access_token=...': {
//       const user = await actions.login(JSON.parse(config.body))
//       return {
//         ok: true,
//         status: 200,
//           json: async () => ({user}),
//         })
//       }
//       const shoppingCart = JSON.parse(config.body)
//       // do whatever other things you need to do with this shopping cart
//       return {
//         ok: true,
//         status: 200,
//         json: async () => ({success: true}),
//       }
//     }
//     default: {
//       throw new Error(`Unhandled request: ${url}`)
//     }
//   }
// }
// beforeAll(() => jest.spyOn(global, 'fetch'))
// beforeEach(() => global.fetch.mockImplementation(mockFetch))
