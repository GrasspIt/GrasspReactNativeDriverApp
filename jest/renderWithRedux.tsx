// https://redux.js.org/recipes/writing-tests
// https://callstack.com/blog/react-native-testing-library-with-redux/

import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react-native';
import { store } from '../store/store';

const renderWithRedux = (component, { reduxStore = store, ...renderOptions } = {}) => {
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return render(component, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { renderWithRedux };
