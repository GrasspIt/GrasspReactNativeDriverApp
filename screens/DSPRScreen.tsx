import React from "react";
import { useSelector } from "react-redux";
import { State } from "../store/reduxStoreState";
import { StackNavigationProp } from "@react-navigation/stack";
import { DrawerStackParamsList } from '../navigation/DrawerNavigator';
import { View, StyleSheet, FlatList, Text } from "react-native";
import Colors from '../constants/Colors';
import DsprCard from "../components/DsprCard";

type DSPRScreenNavigationProp = StackNavigationProp<DrawerStackParamsList, 'DSPRs'>;
type Props = { navigation: DSPRScreenNavigationProp }

const DSPRScreen = ({ navigation }: Props) => {

    const dsprs = useSelector<State, Object>(state => state.api.entities.DSPRs);
    const dsprDrivers = useSelector<State, Object>(state => state.api.entities.dsprDrivers);

    const dsprDataList = Object.values(dsprs);
    let dsprDriverDataList;
    if (dsprDrivers) {
        dsprDriverDataList = Object.values(dsprDrivers);
    }

    const handleSelectDspr = (dsprId: number) => {
        // find the dsprDriver that matches the dsprId
        const selectedDriver = dsprDriverDataList.find((driver: any) => driver.dspr === dsprId);
        // navigate to dashboard passing driverId as props
        navigation.navigate('Dashboard', { driverId: selectedDriver.id });
    }

    return (
        <View style={styles.container}>
            <View style={styles.titleContainer}>
                <Text style={styles.title}>Choose a Dispensary</Text>
            </View>
            <FlatList
                data={dsprDataList}
                renderItem={item => <DsprCard handleSelect={handleSelectDspr} dspr={item.item}/>}
                keyExtractor={(item: any) => item.id.toString()}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    titleContainer: {
        paddingTop: 40,
        alignItems: 'center'
    },
    title: {
        fontSize: 20,
        padding: 10
    },
    container: {
      flex: 1,
      backgroundColor: Colors.light,
      justifyContent: 'center'
    },
})

export default DSPRScreen;