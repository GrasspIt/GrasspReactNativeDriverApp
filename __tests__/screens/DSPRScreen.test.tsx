import React from 'react';
import { renderWithRedux } from '../../jest/renderWithRedux';

import DSPRScreen from '../../screens/DSPRScreen';

describe('<DSPRScreen />', () => {
  it('should render with given state from Redux store', async () => {
    await expect(renderWithRedux(<DSPRScreen />).toJSON()).toMatchSnapshot();
  });
  //   it('should dispatch an action on button click', () => {

  // });
});
