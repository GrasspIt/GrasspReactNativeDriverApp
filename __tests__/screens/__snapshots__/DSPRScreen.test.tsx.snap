// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DSPRScreen /> renders with given state from Redux store 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 22,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Choose a Dispensary
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<DSPRScreen /> should render with given state from Redux store 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 22,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Choose a Dispensary
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
